### ISSUE TITLE
[Bug] Upload to the container registry is not authenticated

### ISSUE THREAD

**Issue by mnitchev** [cloudfoundry/cf-k8s-controllers#613]
While implementing #545 we noticed that even if the Patch that sets the ImageRef on the Package fails with 403 Forbidden, the image is already uploaded to the registry. This means tha
t unauthorized users can still **actually* upload the package even though they don't have permission to do that.

Given that the container registry does not use the Kubernetes RBAC we need to do this authorization ourselves. One way would be to test if the user has Patch permissions on the Packag
e resource before we do the image upload (simillar to how `kubectl auth can-i` works; see [here](https://kubernetes.io/docs/reference/access-authn-authz/authorization/#checking-api-ac
cess))

**Comment by tcdowney** [comment]
Yeah permissions check on the ability to `PATCH`/`PUT` a `Package` before uploading is definitely required. That `kubectl auth can-i` looks cool.
